src/fluree/db/ledger/storage/s3store.clj

```diff
@@ -94,9 +94,8 @@
                      req)
         resp       (aws/invoke client req)
         _          (when (:cognitect.anomalies/category resp)
-                     (if-let [err (:cognitect.aws.client/throwable resp)]
-                       err
-                       (ex-info "S3 list failed" {:response resp})))
+                     (throw (or (:cognitect.aws.client/throwable resp)
+                                (ex-info "S3 list failed" {:response resp}))))
```

Comment:
I think we probably want to return the exception instead of throwing it (like the write fn in here does). Since they're typically called async, we need to return the exception instead of throwing it.